### PR TITLE
Fix massflux bug for N_up>1

### PR DESCRIPTION
**Pull request by yairchn**
The current computation of the mass flux is true to what is written in the second row in equation (10) Cohen et al 2020 assuming `i=0` in the first summation and summing over `j` only. To code the second line in this equation for multiple updraft one would have to use two for-loops (for the two summations) and add factor `1/2`. 
Instead I am coding here the first row in equation (10) Cohen et al 2020, by taking the difference from the grid mean for each subdomain and having the correct flux coded with only one for-loop. 
The bug originates in SCAMPy https://github.com/CliMA/SCAMPy/blob/master/Turbulence_PrognosticTKE.pyx

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]
Code changes look good to me.

File: src/EDMF_functions.jl [hunk 1]
```diff
@@ -70,51 +72,58 @@ function compute_sgs_flux!(edmf::EDMFModel, grid::Grid, state::State, surf::Surf
...
-        aux_up_i = aux_up[i]
         θ_liq_ice_up = aux_up_i.θ_liq_ice
         q_tot_up = aux_up_i.q_tot
-        massflux = aux_up_f[i].massflux
+        @. aux_up_f[i].massflux = ρ0_f * Ifau(a_up) * (w_up - w_gm)
         # We know that, since W = 0 at z = 0, m = 0 also, and
         # therefore θ_liq_ice / q_tot values do not matter
-        @. massflux_h += massflux * (If(θ_liq_ice_up) - If(θ_liq_ice_en))
-        @. massflux_qt += massflux * (If(q_tot_up) - If(q_tot_en))
+        @. massflux_h += ρ0_f * (Ifau(a_up) * (w_up - w_gm) * (If(θ_liq_ice_up) - If(θ_liq_ice_en)))
```
Comment (ilopezgp) [inline src/EDMF_functions.jl:100]: if you are adding the contribution over all subdomains, you should subtract the gmv values to the up values, not the en values.

File: src/EDMF_functions.jl [hunk 2]
```diff
@@ -70,51 +72,58 @@ function compute_sgs_flux!(edmf::EDMFModel, grid::Grid, state::State, surf::Surf
...
+        @. massflux_qt += ρ0_f * (Ifau(a_up) * (w_up - w_gm) * (If(q_tot_up) - If(q_tot_en)))
```
Comment (ilopezgp) [inline src/EDMF_functions.jl:101]: Same here

File: src/EDMF_functions.jl [hunk 3]
```diff
@@ -70,51 +72,58 @@ function compute_sgs_flux!(edmf::EDMFModel, grid::Grid, state::State, surf::Surf
...
             aux_up_f_i = aux_up_f[i]
             aux_up_i = aux_up[i]
             q_liq_up = aux_up_i.q_liq
             q_ice_up = aux_up_i.q_ice
-            massflux = aux_up_f[i].massflux
-            @. massflux_ql += massflux * (If(q_liq_up) - If(q_liq_en))
-            @. massflux_qi += massflux * (If(q_ice_up) - If(q_ice_en))
+            massflux_up_i = aux_up_f[i].massflux
+            @. massflux_ql += massflux_up_i * (w_up - w_gm) * (If(q_liq_up) - If(q_liq_gm))
+            @. massflux_qi += massflux_up_i * (w_up - w_gm) * (If(q_ice_up) - If(q_ice_gm))
```
Comment (ilopezgp) [inline src/EDMF_functions.jl:123]: This doesn't seem right. You are multiplying a mass flux by vertical velocity again.

**Comment by ilopezgp** [comment]
Apart from the technicalities, I am all in favor to change to the first line of eqn (10), instead of the second.

**Review by ilopezgp — APPROVED** [review]
LGTM

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**